lib/yang_wrappers.c

```diff
@@ -1137,6 +1138,30 @@ void yang_str2mac(const char *value, struct ethaddr *mac)
     (void)prefix_str2mac(value, mac);
 }
 
+struct yang_data *yang_data_new_date_and_time(const char *xpath, time_t time)
+{
+	struct tm tm;
+	char timebuf[MONOTIME_STRLEN];
+	struct timeval _time, time_real;
+	char *ts_dot;
+
+	_time.tv_sec = time;
+	_time.tv_usec = 0;
+	monotime_to_realtime(&_time, &time_real);
+
+	gmtime_r(&time_real.tv_sec, &tm);
+
+	/* rfc-3339 format */
+	strftime(timebuf, sizeof(timebuf), "%Y-%m-%dT%H:%M:%S", &tm);
+	ts_dot = timebuf + strlen(timebuf);
+
+	/* microseconds and appends Z */
+	snprintfrr(ts_dot, strlen(timebuf), ".%06luZ",
```

Comment:
Err... wtf is the `strlen(timebuf)` doing here?  That's `sizeof(timebuf) - strlen(timebuf)` if anything.

Reply:
oops missed one while approving.

Reply:
Ack @eqvinox `ts_dot` length is part of the original buffer (timebuf). I have corrected it in latest patch.